src/libraries/System.Formats.Tar/tests/TarEntry/TarEntryGnu.Tests.cs

```diff
@@ -38,6 +38,220 @@ public void Constructor_UnsupportedEntryTypes()
             Assert.Throws<InvalidOperationException>(() => new GnuTarEntry(TarEntryType.LongPath, InitialEntryName));
         }
 
+        [Fact]
+        public void Constructor_ConversionFromV7()
+        {
+            V7TarEntry v7 = new V7TarEntry(TarEntryType.V7RegularFile, InitialEntryName);
+            GnuTarEntry convertedV7 = new GnuTarEntry(other: v7);
+
+            Assert.Equal(TarEntryType.RegularFile, convertedV7.EntryType);
+            Assert.Equal(InitialEntryName, convertedV7.Name);
+        }
+
+        [Fact]
+        public void Constructor_ConversionFromUstar()
+        {
+            UstarTarEntry ustar = new UstarTarEntry(TarEntryType.RegularFile, InitialEntryName);
+            GnuTarEntry convertedUstar = new GnuTarEntry(other: ustar);
+
+            Assert.Equal(TarEntryType.RegularFile, convertedUstar.EntryType);
+            Assert.Equal(InitialEntryName, convertedUstar.Name);
+        }
```

Comment:
There's a lot more that feels like it can/should be asserted in these tests.

e.g., for this one:

```suggestion
        public void Constructor_ConversionFromUstar()
        {
            UstarTarEntry ustar = new UstarTarEntry(TarEntryType.RegularFile, InitialEntryName);
            ustar.ModificationTime = new DateTimeOffset(2000, 1, 21, 15, 23, 51, 922, TimeSpan.Zero);
            // maybe set UID/GID/etc
            DateTimeOffset now = DateTimeOffset.UtcNow;
            GnuTarEntry convertedUstar = new GnuTarEntry(other: ustar);

            Assert.Equal(TarEntryType.RegularFile, convertedUstar.EntryType);
            Assert.Equal(InitialEntryName, convertedUstar.Name);
            Assert.Equal(ustar.ModificationTime, convertedUstar.ModificationTime);

            // ctime and atime are set to the time the constructor was called, UTC
            Assert.Equal(convertedUstar.AccessTime, convertedUstar.CreatedTime);
            Assert.Equal(TimeSpan.Zero, convertedUstar.AccessTime);
            Assert.NotEqual(ustar.ModificationTime, convertedUstar.AccessTime);
            
            // Allow up to 5 seconds of clock drift between capture time and the constructor call,
            // that should (hopefully) keep test noise to a minimum.
            Assert.Equal(now, convertedUstar.AccessTime, TimeSpan.FromSeconds(5));
            
            // UID/GID/etc.
        }
```

Reply:
Code simplified.

Reply:
Fixed this code to be reused and also to verify all the expected fields to convert.